main.nf

```diff
@@ -107,81 +114,97 @@ params.gff = params.genome ? params.genomes[ params.genome ].gff ?: false : fals
 }
+if (params.pseudo_aligner && params.pseudo_aligner != 'salmon'){
+    exit 1, "Invalid pseudo aligner option: ${params.pseaudo_aligner}. Valid options: 'salmon'"
+}
 if( params.star_index && params.aligner == 'star' ){
     star_index = Channel
-        .fromPath(params.star_index)
+        .fromPath(params.star_index, checkIfExists: true)
         .ifEmpty { exit 1, "STAR index not found: ${params.star_index}" }
 }
 else if ( params.hisat2_index && params.aligner == 'hisat2' ){
     hs2_indices = Channel
-        .fromPath("${params.hisat2_index}*")
+        .fromPath("${params.hisat2_index}*", checkIfExists: true)
         .ifEmpty { exit 1, "HISAT2 index not found: ${params.hisat2_index}" }
 }
 else if ( params.fasta ){
-    Channel.fromPath(params.fasta)
-           .ifEmpty { exit 1, "Fasta file not found: ${params.fasta}" }
-           .into { ch_fasta_for_star_index; ch_fasta_for_hisat_index}
+    Channel.fromPath(params.fasta, checkIfExists: true)
+        .ifEmpty { exit 1, "Genome fasta file not found: ${params.fasta}" }
+        .into { ch_fasta_for_star_index; ch_fasta_for_hisat_index; ch_fasta_for_salmon_transcripts }
 }
 else {
-    exit 1, "No reference genome specified!"
+    exit 1, "No reference genome files specified!"
+}
+
+if( params.aligner == 'hisat2' && params.splicesites ){
+    Channel
+        .fromPath(params.bed12, checkIfExists: true)
+        .ifEmpty { exit 1, "HISAT2 splice sites file not found: $alignment_splicesites" }
+        .into { indexing_splicesites; alignment_splicesites }
+}
+
+if ( params.pseudo_aligner == 'salmon' ) {
+    if ( params.salmon_index ) {
+        salmon_index = Channel
+            .fromPath(params.salmon_index, checkIfExists: true)
+            .ifEmpty { exit 1, "Salmon index not found: ${params.salmon_index}" }
+    } else if ( params.transcript_fasta ) {
+        ch_fasta_for_salmon_index = Channel
+            .fromPath(params.transcript_fasta, checkIfExists: true)
+            .ifEmpty { exit 1, "Transcript fasta file not found: ${params.transcript_fasta}" }
+    }
 }
 
 if( params.gtf ){
     Channel
-        .fromPath(params.gtf)
+        .fromPath(params.gtf, checkIfExists: true)
         .ifEmpty { exit 1, "GTF annotation file not found: ${params.gtf}" }
-        .into { gtf_makeSTARindex; gtf_makeHisatSplicesites; gtf_makeHISATindex; gtf_makeBED12;
-              gtf_star; gtf_dupradar; gtf_qualimap;  gtf_featureCounts; gtf_stringtieFPKM }
-} else if( params.gff ){
-  gffFile = Channel.fromPath(params.gff)
-                   .ifEmpty { exit 1, "GFF annotation file not found: ${params.gff}" }
+        .into { gtf_makeSTARindex; gtf_makeHisatSplicesites; gtf_makeHISATindex; gtf_makeSalmonIndex; gtf_makeBED12;
+                gtf_star; gtf_dupradar; gtf_qualimap;  gtf_featureCounts; gtf_stringtieFPKM; gtf_salmon; gtf_salmon_merge }
+} else if ( params.gff ){
+    gffFile = Channel.fromPath(params.gff, checkIfExists: true)
```

Comment:
@kerimoff I think it is just sylistic. I tend to use `gffFile = ` when you only need a single channel for the downstream processes because it saves one line of code 😜 I try to only use `set ( gffFile )` if I need to change the name of a channel because Ive manipulated its contents.

Could use `.set( gffFile )` too 👍 but all of these dilemma's should disappear when we migrate to `DSL2` anyway...